homeassistant/util/yaml.py

```diff
@@ -162,18 +160,22 @@ def _secret_yaml(loader: SafeLineLoader,
                                   " but 'logger: %s' found", logger)
                 del secrets['logger']
         else:
-            loader._SECRET_CACHE[secret_path] = None
-    secrets = loader._SECRET_CACHE[secret_path]
+            _SECRET_CACHE[secret_path] = {}
+    secrets = _SECRET_CACHE[secret_path]
 
     # Retrieve secret, first from secrets.yaml, then from keyring
     if secrets is not None and node.value in secrets:
         _LOGGER.debug('Secret %s retrieved from secrets.yaml.', node.value)
         return secrets[node.value]
-    for sname, sdict in loader._SECRET_CACHE.items():
-        if node.value in sdict:
+
+    while os.path.exists(secret_path) and secret_path in _SECRET_CACHE:
```

Comment:
`hass.config.config_dir` but we are breaking the rule that a util is not home-assistant specific

`os.path.dirname(sys.path[0])` will be the top config path due to deps being loaded from bootstrap (see link below, couple of lines up)

Also, add a call to your clear_passwords function [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/bootstrap.py#L311) to clear it after loading